Internal signal refactors

crisbeto opened the pull request:
Reorganizes some signal-related code to make it easier to reuse.

crisbeto committed:
refactor(core): move input signals into own directory

Moves the signal input code into its own directory to avoid too many files in the `authoring` root once model inputs are introduced.

crisbeto committed:
refactor(compiler-cli): allow signal input transform to be reused

Splits up the signal input transform so that it can be reused for other signal-based properties.

devversion commented on packages/compiler-cli/src/transformers/jit_transforms/signal_metadata_transform_factory.ts:
```diff
@@ -0,0 +1,99 @@
+/** Decorators for classes that should be transformed. */
+const decoratorsWithInputs = ['Directive', 'Component'];
+
+/** Function that can be used to transform a class property to add signal-specific metadata. */
```
Nit: Would be great to rename this to something for "initializer-based APIs` as output for example will not use signals, and needs a similar transform.

devversion committed:
refactor(compiler-cli): group initializer-API based transforms into single transform

Instead of maintaining individual transforms for `input`, `output`,
`model` etc. we are grouping them directly and the first one matching,
will execute.

This reduces needed traversal through AST and also makes it a little
more clean to write new initializer API metadata transforms.

Note: The Angular JIT transform is now also moving from `tooling.ts`
directly into `/transformers` for more local placement of transformer
logic.

devversion commented:
Note: Non-public facing change of the JIT transform, so we can skip the required explicit reviews from tooling side.

alan-agius4 reviewed (approved):
Reviewed of #tooling-cli-shared-api

devversion committed:
fixup! refactor(compiler-cli): group initializer-API based transforms into single transform

Continue exposing JIT angular transform for saucelabs job/advanced customers like Jest

crisbeto reviewed (commented):
I can't approve my own PR, but LGTM on Paul's changes.

jessicajaniuk commented:
This PR was merged into the repository by commit d74ee6e343ef319acb33d5e05bf96b3150050266.
